jgrapht-core/src/test/java/org/jgrapht/alg/decomposition/HeavyPathDecompositionTest.java

```diff
@@ -64,6 +65,7 @@ public static int log2(int n) // returns 0 for n=0
         }
 
         int maxim = 0;
+        HeavyPathDecomposition.InternalState state = decomposition.getInternalState();
```

Comment:
If you change this to HeavyPathDecomposition<V,E>.InternalState, won't that make your typecasting woes go away?

Reply:
I need to stop using Intellij's autocomplete blindly. Thank you, @jsichi!

Reply:
I am glad we didn't have to resort to reflection :)